nautobot/docs/installation/region-and-site-data-migration-guide.md

```diff
@@ -0,0 +1,408 @@
+
+```python
+# models.py
+class ExampleModel(OrganizationalModel):
+    # note that site field is gone
+    location = models.ForeignKey(to="dcim.Location", on_delete=models.PROTECT, blank=True, null=True)
+    name = models.CharField(max_length=20, help_text="The name of this Example.")
+...
+```
+
+After removing the `site` attribute, make the migration file by running `nautobot-server makemigrations [app_name] -n [migration_name]`, for example:
+
+```shell
+nautobot-server makemigrations example_app -n remove_site_field_from_example_model
+```
+
+The migration file might look like this:
+
+```python
+# Generated by Django 3.2.17 on 2023-02-22 17:09
+# 0009_remove_site_field_from_example_model.py
+
+from django.db import migrations
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('example_app', '0008_migrate_example_model_data_from_site_to_location'),
+    ]
+
+    operations = [
+        migrations.RemoveField(
+            model_name='examplemodel',
+            name='site',
+        ),
+    ]
+```
+
+!!! important
+    Before you apply this migration, you have to add a `run_before` attribute in this migration file to make sure that you remove `site`/`region` fields before `Site` and `Region` models themselves are removed. **Without it, your migration files might be out of order and your app will not start**.
+
+```python
+    # Ensure this migration is run before the migration that removes Region and Site Models
+    run_before = [
+        # TODO we need to change the name after we reorder migration files
+        ("dcim", "0034_remove_region_and_site"),
+    ]
+```
+
+The final migration file might look like this:
+
+```python
+# Generated by Django 3.2.17 on 2023-02-22 17:09
+# 0009_remove_site_field_from_example_model.py
+
+from django.db import migrations
+
+class Migration(migrations.Migration):
+    
+    # Ensure this migration is run before the migration that removes Region and Site Models
+    run_before = [
+        # TODO we need to change the name after we reorder migration files
+        ("dcim", "0034_remove_region_and_site"),
+    ]
+    dependencies = [
+        ("example_app", "0008_migrate_example_model_data_from_site_to_location"),
+    ]
+
+    operations = [
+        migrations.RemoveField(
+            model_name="examplemodel",
+            name="site",
+        ),
+    ]
+```
+
+!!!important
+    Before you apply the aforementioned migration files to your app in the following step, please make sure that Nautobot is on version 1.x and not on version 2.0. More importantly, please make sure that `("dcim", "0030_migrate_region_and_site_data_to_locations")` is applied to your Nautobot instance and that `("dcim", "0034_remove_region_and_site")` is **not** applied.
+
```

Comment:
Would be good to flesh this out a bit further with exact steps, something like:

- Start with a working Nautobot 1.5.x installation with baseline migrations all run
- Stop Nautobot server
- Update installed version of Nautobot to 2.0.0
- Run `nautobot-server migrate dcim 0030_migrate_region_and_site_data_to_locations` (?)
- Updated installed version of *all* Apps to include any necessary migration files as described above
- For each affected app, run `nautobot-server migrate <app_name>` (?)
- Run `nautobot-server migrate` to finish all migrations (?)
- Start Nautobot server

Reply:
I moved this to before the guide starts